department-of-veterans-affairs/caseflow
# Adding POA and AOD to AMA case details

Connects #6045

### Description
Three changes:
1) Added advanced_on_docket to the DB, and return that in the appeal serializer.
2) Removed RO from appeal serializer since AMA appeals don't really have one.
3) Refactored the PowerOfAttorney model to use a new BgsPowerOfAttorney service. We delegate a lot of methods there so that AMA appeals can also use the service. 

![screen shot 2018-06-29 at 1 14 17 pm](https://user-images.githubusercontent.com/3885236/42105445-5e3e10f4-7b9e-11e8-9be4-42c6ea570d97.png)

### Acceptance Criteria 
- [ ] AOD and POA are properly displayed on queue details screen for AMA appeals. RO is not shown.

### Testing Plan
1. Make sure :queue_beaam_appeals is enabled.
1) Run rake db:seed
1) Go to queue/beaam
1) Click into an appeal and ensure that AOD and POA are properly displayed.
1) Ensure that RO is not displayed
1) Ensure that when navigating to Reader and clicking on the claim details dropdown that RO is not displayed.
